backend/src/sms/services/sms-callback.service.ts

```diff
@@ -23,7 +24,21 @@ const isAuthenticated = (
   const [username, password] = credentials.split(':')
   const plainTextPassword =
     username + messageId + campaignId + config.get('smsCallback.callbackSecret')
-  return bcrypt.compareSync(plainTextPassword, password)
+
+  let isAuthenticated = getSha256Hash(plainTextPassword) === password
```

Comment:
```suggestion
  let isAuthenticated = crypto.timingSafeEqual(
      getSha256Hash(plainTextPassword),
      password
  )
```

Though very unlikely, when doing string equality comparison for crypto, should use [constant time comparison functions](https://nodejs.org/docs/latest-v16.x/api/crypto.html#cryptotimingsafeequala-b) to prevent [timing attacks](https://paragonie.com/blog/2015/11/preventing-timing-attacks-on-string-comparison-with-double-hmac-strategy).